packages/flutter/lib/src/material/tabs.dart

```diff
@@ -568,11 +568,108 @@ class _TabBarScrollController extends ScrollController {
 ///
 /// Uses values from [TabBarTheme] if it is set in the current context.
 ///
-/// To see a sample implementation, visit the [TabController] documentation.
+/// {@tool dartpad --template=stateless_widget_material}
+/// This sample shows the implementation of [TabBar] and [TabBarView] using a [DefaultTabController].
+/// Each [Tab] corresponds to a child of the [TabBarView] in the order they are written.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///    return DefaultTabController(
+///      initialIndex: 1,
+///      length: 3,
+///      child: Scaffold(
+///        appBar: AppBar(
+///          title: Text('TabBar Widget'),
+///          bottom: TabBar(
+///            tabs: <Widget>[
+///              Tab(
+///                icon: Icon(Icons.cloud_outlined),
+///              ),
+///              Tab(
+///                icon: Icon(Icons.beach_access_sharp),
+///              ),
+///              Tab(
+///                icon: Icon(Icons.brightness_5_sharp),
+///              ),
+///            ],
+///          ),
+///        ),
+///        body: TabBarView(
+///          children: <Widget>[
+///            Center(
+///              child: Text('It\'s cloudy here'),
+///            ),
+///            Center(
+///              child: Text('It\'s rainy here'),
+///            ),
+///            Center(
+///              child: Text('It\'s sunny here'),
+///            ),
+///          ],
+///        ),
+///      ),
+///    );
+///  }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
```

Comment:
There is another template just for this case! It takes care of the mixin. :)
```suggestion
/// {@tool dartpad --template=stateful_widget_material_ticker}
```